src/map.cpp

```diff
@@ -532,62 +532,150 @@ bool Map::canThrowObjectTo(const Position& fromPos, const Position& toPos, bool
 	return isSightClear(fromPos, toPos, false);
 }
 
-bool Map::checkSightLine(const Position& fromPos, const Position& toPos) const
+//get a line between two points
+//based on Xiaolin Wu algorithm
+std::vector< std::pair<int, int> > get2DLine(int x0, int y0, int x1, int y1)
 {
-	if (fromPos == toPos) {
-		return true;
+	std::vector< std::pair<int, int> > line;
+	if (x0 == x1 && y0 == y1) {
+		return line;
 	}
 
-	Position start(fromPos.z > toPos.z ? toPos : fromPos);
-	Position destination(fromPos.z > toPos.z ? fromPos : toPos);
+	bool isSteep = std::abs(y1 - y0) > std::abs(x1 - x0);
 
-	const int8_t mx = start.x < destination.x ? 1 : start.x == destination.x ? 0 : -1;
-	const int8_t my = start.y < destination.y ? 1 : start.y == destination.y ? 0 : -1;
+	if (isSteep) {
+		std::swap(x0, y0);
+		std::swap(x1, y1);
+	}
 
-	int32_t A = Position::getOffsetY(destination, start);
-	int32_t B = Position::getOffsetX(start, destination);
-	int32_t C = -(A * destination.x + B * destination.y);
+	if (x0 > x1) {
+		std::swap(x0, x1);
+		std::swap(y0, y1);
+	}
```

Comment:
Tip: you can use [std::minmax](https://en.cppreference.com/w/cpp/algorithm/minmax) to achieve the same result without the explicit branch:

```cpp
std::tie(x0, x1) = std::minmax(x0, x1);
std::tie(y0, y1) = std::minmax(y0, y1);
```
And then `x0` will always be less or equal than `x1`, ditto for `y0` and `y1`.

Reply:
I tried to implement this, it could be because i'm in over my head with it, but it wouldn't produce the same outcome.